controllers/mover/syncthing/syncthing_test.go

```diff
@@ -474,7 +477,173 @@ var _ = Describe("When an RS specifies Syncthing", func() {
+						},
+					}
+					err = mover.ensureStatusIsUpdated(service)
+					Expect(err).ToNot(BeNil())
+					Expect(err.Error()).To(ContainSubstring("no such host"))
+					Expect(err.Error()).To(ContainSubstring("Get"))
+
+					_, err = mover.syncthing.jsonRequest("/rest/config", "GET", nil)
+					Expect(err).ToNot(BeNil())
+					Expect(err.Error()).To(ContainSubstring("no such host"))
+					Expect(err.Error()).To(ContainSubstring("Get"))
+				})
+			})
+
+			When("Syncthing server exists", func() {
+				var ts *httptest.Server
+				var s SyncthingConfig
+
+				JustBeforeEach(func() {
+					sConfig := SyncthingConfig{
+						Version: 10,
+					}
+					sStatus := SystemStatus{
+						MyID: "test",
+					}
+
+					sConnections := SystemConnections{
+						Total: TotalStats{At: "test"},
+					}
+
+					ts = httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+						switch r.URL.Path {
+						case "/rest/config":
+							if r.Method == "GET" {
+								res := sConfig
+								resBytes, _ := json.Marshal(res)
+								fmt.Fprintln(w, string(resBytes))
+							} else if r.Method == "PUT" {
+								err := json.NewDecoder(r.Body).Decode(&s)
+								if err != nil {
+									http.Error(w, "Error decoding request body", http.StatusBadRequest)
+									return
+								}
+							}
+							return
+						case "/rest/system/status":
+							res := sStatus
+							resBytes, _ := json.Marshal(res)
+							fmt.Fprintln(w, string(resBytes))
+							return
+						case "/rest/system/connections":
+							res := sConnections
+							resBytes, _ := json.Marshal(res)
+							fmt.Fprintln(w, string(resBytes))
+							return
+						default:
+							return
+						}
+					}))
+
+					mover.syncthing.APIConfig.APIURL = ts.URL
+					mover.syncthing.APIConfig.Client = ts.Client()
+					mover.syncthing.APIConfig.APIKey = "test"
+				})
+
+				JustAfterEach(func() {
+					ts.Close()
+				})
+
+				It("Fetches the Latest Info", func() {
+					err := mover.syncthing.FetchLatestInfo()
+					Expect(err).To(BeNil())
+					Expect(mover.syncthing.Config.Version).To(Equal(10))
+					Expect(mover.syncthing.SystemStatus.MyID).To(Equal("test"))
+					Expect(mover.syncthing.SystemConnections.Total.At).To(Equal("test"))
+				})
+
+				It("Updates the Syncthing Config", func() {
+					mover.syncthing.Config = &SyncthingConfig{
+						Version: 9,
+					}
+					err := mover.syncthing.UpdateSyncthingConfig()
+					Expect(err).To(BeNil())
+					Expect(s.Version).To(Equal(9))
+				})
+
+				It("Ensures it's configured", func() {
+					apiKeys := &corev1.Secret{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "volsync-" + mover.owner.GetName(),
+							Namespace: ns.Name,
+						},
+						Data: map[string][]byte{
+							"apikey":   []byte("my-secret-apikey-do-not-steal"),
+							"username": []byte("gcostanza"),
+							"password": []byte("bosco"),
+						},
+					}
+					err := mover.ensureIsConfigured(apiKeys)
```

Comment:
We'll want to test that this is working by seeing if the changes made by VolSync exist in our mock API. You can accomplish this by adding an entry to `mover.peerList` and validating that the mocked server received it.